### PR TITLE
Search installation folder for plugins after everything else fails.

### PR DESCRIPTION
**Pull request by takeda**
This PR makes serverless search plugins in installation folder when it can't find them anywhere else.

Closes: #7976

### CODE REVIEW

File: lib/classes/PluginManager.js [hunk 1]
```diff
@@ -23,7 +23,14 @@ const requireServicePlugin = (servicePath, pluginPath, localPluginsPath) => {
...
   }
-  return require(cjsResolve(servicePath, pluginPath).realPath);
+  try {
+    return require(cjsResolve(servicePath, pluginPath).realPath);
+  } catch (error) {
+    if (!isModuleNotFoundError(error, pluginPath)) throw error;
+  }
+
+  // as a last ditch effort try location where serverless started from
+  return require(cjsResolve(path.dirname(require.main.filename), pluginPath).realPath);
```
Comment (medikoo) [inline lib/classes/PluginManager.js:33]: I think it's fine to simple use `__dirname` (as `require.main` does not necessarily may point expected path)

File: lib/classes/PluginManager.js [hunk 2]
```diff
@@ -23,7 +23,14 @@ const requireServicePlugin = (servicePath, pluginPath, localPluginsPath) => {
...
     }
   }
-  return require(cjsResolve(servicePath, pluginPath).realPath);
+  try {
+    return require(cjsResolve(servicePath, pluginPath).realPath);
+  } catch (error) {
+    if (!isModuleNotFoundError(error, pluginPath)) throw error;
+  }
+
+  // as a last ditch effort try location where serverless started from
```
Comment (medikoo) [inline lib/classes/PluginManager.js:32]: Let's reword it it to: _Search in "node_modules" in which framework is placed_

**Comment by takeda** [comment]
@medikoo I applied mentioned suggestions (apologies it took me that long, but I was busy with work)

File: lib/classes/PluginManager.js [hunk 3]
```diff
@@ -23,7 +23,14 @@ const requireServicePlugin = (servicePath, pluginPath, localPluginsPath) => {
       if (!isModuleNotFoundError(error, absoluteLocalPluginPath)) throw error;
     }
   }
-  return require(cjsResolve(servicePath, pluginPath).realPath);
+  try {
+    return require(cjsResolve(servicePath, pluginPath).realPath);
+  } catch (error) {
+    if (!isModuleNotFoundError(error, pluginPath)) throw error;
```
Comment (medikoo) [inline lib/classes/PluginManager.js:29]: Sorry just noticed, We should add following to condition ` || pluginPath.startsWith('.')`.

As global `node_modules` search should occur only for _package_ requires (not relative modules)

**Comment by takeda** [comment]
@medikoo applied suggestions

**Review by medikoo — APPROVED** [review]
Thank you @takeda!

**Pull request merged**